### PR TITLE
Migrate the rest of package:dwds to null safety

### PR DESCRIPTION
**Pull request by annagrin**
Migrate the rest of package:dwds to null safety

- handlers
  - dev_handler.dart
- lib 
  - dart_web_debug_service.dart  
  - dwds.dart

Also, added defaults to all parameters in test setup and removed unused one (discovered nulls passed as parameters due to migration)

Towards: https://github.com/dart-lang/webdev/issues/1327

### CODE REVIEW

**Review by grouma — APPROVED** [review]

File: dwds/lib/dart_web_debug_service.dart [hunk 1]
```diff
@@ -67,53 +64,42 @@ class Dwds {
...
+    UrlEncoder? urlEncoder,
+    bool spawnDds = true,
     // TODO(elliette): DevTools is inconsistently capitalized throughout this
-    // file. Change all occurances of devtools/Devtools to devTools/DevTools.
-    bool enableDevtoolsLaunch,
-    DevtoolsLauncher devtoolsLauncher,
-    bool launchDevToolsInNewWindow,
-    SdkConfigurationProvider sdkConfigurationProvider,
-    bool emitDebugEvents,
+    // file. Change all occurences of devtools/Devtools to devTools/DevTools.
```
Comment (elliette) [inline dwds/lib/dart_web_debug_service.dart:83]: Haha we are both wrong 🤦‍♀️ Looked it up and it's spelled "occurrences" (two Rs)

**Review by elliette — APPROVED** [review]
Woohoo! 🎉

File: dwds/lib/dart_web_debug_service.dart (hunk 1 above)
Comment (grouma) [reply dwds/lib/dart_web_debug_service.dart:83]: Assume half the spelling mistakes in this package are mine lol. I can't spell and often forget to enable the spell checker.

Comment (annagrin) [reply dwds/lib/dart_web_debug_service.dart]: We all contribute:) 
I run a spell checker to please the g3 spell checker rules.

**Pull request merged**